install/minikube-installation-guide.md

```diff
@@ -0,0 +1,236 @@
+
+```sh
+minikube ssh
+
+            _         _ ( )           ( )           
+  ___ ___  (_)  ___  (_)| |/')  _   _ | |_      __  
+/' _ ` _ `\| |/' _ `\| || , <  ( ) ( )| '_`\  /'__`\
+| ( ) ( ) || || ( ) || || |\`\ | (_) || |_) )(  ___/
+(_) (_) (_)(_)(_) (_)(_)(_) (_)`\___/'(_,__/'`\____)
+
+# No color option, as minikube is running busybox
+$ egrep 'vmx|svm' /proc/cpuinfo
+# if you get a long line of output here, you have it enabled!
+
+$ exit
+```
+
+## Installing `Kata Containers`
+
+Now you can install the `Kata` runtime components. You will need a local copy of some `Kata`
+components to help with this, and then use the host `kubectl` (that minikube has already
+configured for you) to deploy them:
+
+```sh
+$ git clone https://github.com/kata-containers/packaging.git
+$ cd packaging/kata-deploy
+$ kubectl apply -f kata-rbac.yaml
+$ kubectl apply -f kata-deploy.yaml
+```
+
+This installs the `Kata` components into `/opt/kata` inside the `Minikube` node. It can take
+a few minutes for the operation to complete. Check the installation has worked using:
+
+```sh
+$ minikube ssh
+$ cd /opt/kata
+# If the following directory is not yet created and populated, wait a short while and
+# try again - the download and install process can take a few minutes.
+$ ls bin
+containerd-shim-kata-v2  kata-collect-data.sh  kata-qemu     qemu-ga	     qemu-system-x86_64
+firecracker		 kata-fc	       kata-runtime  qemu-pr-helper  virtfs-proxy-helper
+$ exit
+```
+
+This shows the `Kata` components, including a `qemu`, and the `kata-runtime` for instance.
+
+## Enabling `Kata`
+
+> **Note:** This step is only required for `Minikube`/`k8s` versions <= 1.13. Since version
+> v1.14, the `RuntimeClass` is enabled by default. Performing this step on k8s > v1.14 is
+> however benign.
+
+Now the `Kata` components are installed in the `Minikube` node, you need to configure `k8s`
+`RuntimeClass` to know when to use `Kata` to run a pod.
+
+```sh
+$ curl https://raw.githubusercontent.com/kubernetes/node-api/master/manifests/runtimeclass_crd.yaml > runtimeclass_crd.yaml
+$ kubectl apply -f runtimeclass_crd.yaml
+```
+
+### Register the runtime
+
+Now register the `kata qemu` runtime with that class:
+
+```sh
+$ cd packaging/kata-deploy/k8s-1.14
+$ kubectl apply -f kata-qemu-runtimeClass.yaml
+runtimeclass.node.k8s.io/kata created
+runtimeclass.node.k8s.io/kata-qemu created
+```
+
+`Kata` should now be installed and enabled in the `Minikube` cluster.
+
+## Testing `Kata`
+
+Launch a container that is defined to run on Kata. The enabling is configured by
+the following lines in the yaml file:
+
+```yaml
+    spec:
+      runtimeClassName: kata-qemu
+```
+
+Perform the following action to launch a `Kata` based pod:
+
+```sh
+$ cd packaging/kata-deploy/examples
+$ kubectl apply -f test-deploy-kata-qemu.yaml
+```
+
+An apache php pod should have been deployed with the `Kata` runtime. Wait a few moments
+to check it is running:
+
+```sh
+$ kubectl get pods
```

Comment:
Could this become something like?:

```bash
$ kubectl get pods | grep -- -apache-kata | grep -E "\<Running\>"
```

Reply:
*Surely* there must be some way to detect the pod is running without having to manually eyeball the output though? Maybe a `--json` option or something to make parsing reliable?

Reply:
there is - I'll update to `kubectl rollout status deployment php-apache-kata-qemu`